src/system/SystemLayer.cpp

```diff
@@ -303,6 +303,49 @@ Error Layer::StartTimer(uint32_t aMilliseconds, TimerCompleteFunct aComplete, vo
     return lReturn;
 }
 
+/**
+ * @brief
+ *   This method extends a one-shot timer.
```

Comment:
We should document that if there is no existing matching timer this will just start a new one.  Or document (and implement) that this method errors out if there is no existing timer to be extended...

Is there a reason we chose the "extend or start" behavior rather than the "extend or fail" behavior?

Reply:
See the Note session bellow, lets me know if there is anything unclear. We chose "extend or start" just to align with spec requirement in Arm Fail-safe with these semantics:

"Upon receipt, if there isn’t already an armed fail-safe timer associated with the current secure session, a new fail-safe timer with an expiry time set for a duration of ExpiryLengthSeconds SHALL be armed. Otherwise, the existing fail-safe timer SHALL be extended for a duration of ExpiryLengthSeconds."

Reply:
+1 to @andy31415. I'm surprised why the API doesn't take a timer as an input - or simply just have a call on a timer object to extend?  Having this compare appState feels super strange...